.github/workflows/test.yml

```diff
@@ -137,6 +137,36 @@ jobs:
         if: matrix.version == 'nightly'
         run: just build-bench
 
+  msrv-libs:
```

Comment:
I'm ok with the msrv target, but I think the `just check --all-features` matches this and doesn't require the expansion of all the crate names as it is here, and also will cover the binaries as well. So then we could have just a single `msrv` target.

Reply:
No, because my workflows are testing for different MSRV for libs vs bins.

Reply:
look at #2039, I think that should all all the crates to be on MSRV 1.67

Reply:
I changed my branch to:

* Kill the commit for "Split MSRV testing for libraries and binaries"
* Skip the commit for "use just for the MSRV target"
* Cherry-pick the commit for "add MSRV env-var to justfile for testing"
* Cherry-pick the commit for "update certs for server https/quic tests"
* Squashed the changes from `cargo update -p clap --precise 4.2.7` into "Update semver-compatible dependencies"

Your "revert clap" commit has a lot of unrelated changes to TOML formatting which make it harder to review.